### PR TITLE
add support for image family

### PR DESCRIPTION
**Pull request by vsoch**
I ran into an issue today where Cloud Build was... having a moment?

image

I really needed to enable cgroups v2 to the images (I couldn't do it as a startup script [because reasons](https://twitter.com/jasonmstover/status/1660342910347358216), and I needed cgroups v2 because I wanted to test [k3s in rootless mode](https://docs.k3s.io/advanced#running-rootless-servers-experimental). I was bummed that cloud build wasn't working, but realized that I could instantiate previously built images, do the change to the grub config, and then save as a machine image. The steps looked like this:

1. Went to Compute Engine -> Images
2. Create an instance from each
3. Install cgroups v2 (instructions below)
4. Stop the instances
5. Under Compute Engine -> Image "Create an Image"
6. Create each one from Source Disk from the image Don't forget to use the same prefix for the family (e.g., flux-cgroups-manager-x86-052123 has family flux-cgroups-manager).

Note it's important to set the family for the images as this is how they are selected here. Also note that a machine image is NOT the same as just an image in compute engine (I made this mistake first, oops).

Here are the instructions I used to enable cgroups 2. Note that since I login as my user, sudo is needed. If this is in a root run start script, you likely wouldn't need it.

```bash
# This is what I ran manually to update the machines and save new images
sudo dnf install -y grubby 
sudo grubby --update-kernel=ALL --args="systemd.unified_cgroup_hierarchy=1"

sudo mkdir -p /etc/systemd/system/user@.service.d
cat <<EOF | sudo tee /etc/systemd/system/user@.service.d/delegate.conf
[Service]
Delegate=cpu cpuset io memory pids
EOF
sudo systemctl daemon-reload
```

And then I was able to update the family to be the families I defined here (allowed by the content of this PR) and create the cluster and verify that cgroups 2 is enabled (this was on the login node):

```
cat /sys/fs/cgroup/cgroup.controllers
cpuset cpu io memory hugetlb pids rdma
```

I was able to get k3s (master / control plane) running and am still working on the worker nodes, but I've gotten far enough with this change to open this PR! Feel free to disregard / not add, because it's probably a rare use case to want to modify the family, but I am finding it essential today.

ping @wkharold 

## TLDR

Problem: the user wants to use a custom image family, and currently the families are hard coded.
Solution: add the family name as a variable.

### CODE REVIEW

**Review by wkharold — APPROVED** [review]

**Comment by vsoch** [comment]
Woot!! :partying_face: I love contributing to Google projects on GitHub <3 Thanks @wkharold!
